src/test/java/network/brightspots/rcv/ContestConfigTests.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * RCTab
+ * Copyright (c) 2017-2023 Bright Spots Developers.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Purpose: These tests check configuration files.
+ * Design: Unit tests, and other tests that don't run a tabulation.
+ * Conditions: During automated testing.
+ * Version history: see https://github.com/BrightSpots/rcv.
+ */
+
+package network.brightspots.rcv;
+
+import static network.brightspots.rcv.ContestConfigMigration.isVersionNewer;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class ContestConfigTests {
+  @BeforeAll
+  static void setup() {
+    Logger.setup();
+  }
+
+  @Test
+  @DisplayName("test isVersionNewer")
+  void testVersionComparison() {
+    // Basic checks: version numbers work fine
+    assertFalse(isVersionNewer("1.3.0", "1.3.0"));
+    assertFalse(isVersionNewer("1.2.0", "1.3.0"));
+    assertFalse(isVersionNewer("1.3.0", "1.3.1"));
+    assertTrue(isVersionNewer("1.3.1", "1.3.0"));
+
+    // Version comparison ignores the -alpha specifier
+    assertFalse(isVersionNewer("1.2.0-alpha", "1.3.0"));
+    assertTrue(isVersionNewer("1.4.0-alpha", "1.3.0"));
+    assertFalse(isVersionNewer("1.4.0-alpha", "1.4.0"));
+    assertFalse(isVersionNewer("1.4.0", "1.4.0-alpha"));
+
+    // Sane things with false versions
+    assertFalse(isVersionNewer("goober", "1.0.0"));
+    assertTrue(isVersionNewer("1.0.0", "goober"));
+
+    // Partial versions are parsed correctly and have implicit zeros
+    assertTrue(isVersionNewer("1.0.1", "1.0"));
+    assertFalse(isVersionNewer("1.0.0", "1.0"));
+    assertFalse(isVersionNewer("1.0.0", "1"));
+    assertFalse(isVersionNewer("1", "1.0"));
+
+    // snapshot info is ignored, though it shouldn't be.
+    // The following tests check the current state, though we have an Issue open with
+    // jackson-core to address.
+    // https://github.com/FasterXML/jackson-core/issues/1050
```

Comment:
Could you please add one commented-out versions of the lines we're hoping to be able to change these to once your issue is resolved?

Same applies to many of the tests under this header above as well:

`// Version comparison ignores the -alpha specifier`

Maybe it's worth kind of grouping these two sections together more (looks like you have a couple duplicates up there anyway.

I'd say put all 6 non-duplicated tests between these two sections under the same header here, and add commented-out versions of what the values _should_ be after your issue is resolved.

Reply:
Good catch on the duplicate tests. Removed those.

Added a comment in this section saying the values should all flip from false to true once the issue is resolved.

Reply:
Not all should be true though; just pushed a commit to clarify.